### PR TITLE
Add gpu-id option to select gpu id

### LINKED ISSUE #138: Select GPU which the scripts use
The current test scripts uses mainly gpu=0. It causes memory allocation error when a user runs many processes even if the machine has multiple gpus.
I want to select main GPU ID based on process id.

### PR DESCRIPTION
**Pull request by unnonouno**
fix #138

### CODE REVIEW

File: docker.py [hunk 1]
```diff
@@ -268,8 +269,25 @@ def make_random_name():
                    for i in range(10))
 
 
+def get_num_gpus():
+    out = subprocess.check_output(['nvidia-smi', '-q'])
```
Comment (gwtnb) [inline docker.py]: `nvidia-smi -L` seems better.

File: docker.py [hunk 2]
```diff
@@ -268,8 +269,25 @@ def make_random_name():
...
+        m = re.match(r'Attached GPUs +: (\d+)', line)
+        if m:
+            return int(m.group(1))
+    return 0
+
+
+def select_gpu(offset):
+    num_gpus = get_num_gpus()
+    gpus = range(num_gpus)
+    offset = offset % len(gpus)
```
Comment (gwtnb) [inline docker.py]: `%=`

**Comment by gwtnb** [comment]
LGTM other than above comments.

**Comment by unnonouno** [comment]
I fixed it.

**Pull request merged**